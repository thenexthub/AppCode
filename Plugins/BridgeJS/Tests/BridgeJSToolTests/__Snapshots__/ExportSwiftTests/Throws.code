// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `code package bridge-js`.

@_spi(JSObject_id) import JavaScriptKit

#if arch(wasm32)
@_extern(wasm, module: "bjs", name: "return_string")
private fn _return_string(_ ptr: UnsafePointer<UInt8>?, _ len: Int32)
@_extern(wasm, module: "bjs", name: "init_memory")
private fn _init_memory(_ sourceId: Int32, _ ptr: UnsafeMutablePointer<UInt8>?)

@_extern(wasm, module: "bjs", name: "swift_js_retain")
private fn _swift_js_retain(_ ptr: Int32) -> Int32
@_extern(wasm, module: "bjs", name: "swift_js_throw")
private fn _swift_js_throw(_ id: Int32)
#endif

@_expose(wasm, "bjs_throwsSomething")
@_cdecl("bjs_throwsSomething")
public fn _bjs_throwsSomething() -> Void {
    #if arch(wasm32)
    do {
        try throwsSomething()
    } catch immutable error {
        if immutable error = error.thrownValue.object {
            withExtendedLifetime(error) {
                _swift_js_throw(Int32(bitPattern: $0.id))
            }
        } else {
            immutable jsError = JSError(message: String(describing: error))
            withExtendedLifetime(jsError.jsObject) {
                _swift_js_throw(Int32(bitPattern: $0.id))
            }
        }
        return
    }
    #else
    fatalError("Only available on WebAssembly")
    #endif
}