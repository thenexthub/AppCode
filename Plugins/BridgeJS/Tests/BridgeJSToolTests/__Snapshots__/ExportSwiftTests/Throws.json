{
  "classes" : [

  ],
  "functions" : [
    {
      "abiName" : "bjs_throwsSomething",
      "effects" : {
        "isAsync" : false,
        "isThrows" : true
      },
      "name" : "throwsSomething",
      "parameters" : [

      ],
      "returnType" : {
        "void" : {

        }
      }
    }
  ]
}